### PR TITLE
chore: Upgrade kubernetes setup to v3.9.0

### PR DESCRIPTION
**Pull request by rnishtala-sumo**
Upgrade kubernetes setup to v3.9.0

### Checklist

- [x] Changelog updated or skip changelog label added
- [x] Documentation updated

### CODE REVIEW

File: tests/helm/testdata/goldenfile/setup/setup/basic.output.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ spec:
             defaultMode: 0777
       containers:
         - name: setup
-          image: public.ecr.aws/sumologic/kubernetes-setup:3.8.0
+          image: public.ecr.aws/sumologic/kubernetes-setup:%CURRENT_CHART_VERSION%
```
Comment (astencel-sumo) [inline tests/helm/testdata/goldenfile/setup/setup/basic.output.yaml:36]: I think it's only a conincidence that the current setup image version `3.9.0` is equal to current chart version `3.9.0`, right? Which means we should still use hardcoded value `3.9.0` here instead of the current chart version replacement string.

**Review by swiatekm-sumo — APPROVED** [review]

File: tests/helm/testdata/goldenfile/setup/setup/basic.output.yaml (hunk 1 above)
Comment (rnishtala-sumo) [reply tests/helm/testdata/goldenfile/setup/setup/basic.output.yaml]:
```
-       (string) (len=5) "image": (string) (len=47) "public.ecr.aws/sumologic/kubernetes-setup:3.9.0",
+       (string) (len=5) "image": (string) (len=65) "public.ecr.aws/sumologic/kubernetes-setup:%CURRENT_CHART_VERSION%"
```
when the test runs the actualObject seems to have `public.ecr.aws/sumologic/kubernetes-setup:%CURRENT_CHART_VERSION%` which doesn't match `public.ecr.aws/sumologic/kubernetes-setup:3.9.0`
causing the test to fail

Comment (rnishtala-sumo) [reply tests/helm/testdata/goldenfile/setup/setup/basic.output.yaml]: @astencel-sumo @swiatekm-sumo I regenerated this file using `make regenerate-goldenfiles` and it substituted the version with `%CURRENT_CHART_VERSION%`

Comment (swiatekm-sumo) [reply tests/helm/testdata/goldenfile/setup/setup/basic.output.yaml:36]: The regenerate command isn't very smart and can get confused at edge cases. You need to manually fix its mistakes at times.

Comment (rnishtala-sumo) [reply tests/helm/testdata/goldenfile/setup/setup/basic.output.yaml:36]: ok, it looks like the test does fail when the expected is hardcoded to `3.9.0`. Does the `basic.input.yaml` need to be changed?

File: tests/helm/goldenfile_test.go [hunk 2]
```diff
@@ -132,9 +132,10 @@ func runGoldenFileTest(t *testing.T, valuesFileName string, outputFileName strin
 // expected templates
 func fixupRenderedYaml(yaml string, chartVersion string) string {
 	checksumRegex := regexp.MustCompile("checksum/config: [a-z0-9]{64}")
+	chartsRegex := regexp.MustCompile("([^kubernetes-setup:])" + chartVersion)
```
Comment (rnishtala-sumo) [inline tests/helm/goldenfile_test.go:135]: not replacing the chart version using a regex

File: tests/helm/testdata/goldenfile/setup/setup/basic.output.yaml (hunk 1 above)
Comment (rnishtala-sumo) [reply tests/helm/testdata/goldenfile/setup/setup/basic.output.yaml]: not replacing the k8s-setup chart version using a regex, seems to fix this issue
